awx/main/scheduler/task_manager.py

```diff
@@ -53,103 +53,86 @@ def inner(*args, **kwargs):
-            # if we detect a failed or error dependency, go ahead and fail this
-            # task. The errback on the dependency takes some time to trigger,
-            # and we don't want the task to enter running state if its
-            # dependency has failed or errored.
-            elif dep.status in ("error", "failed"):
-                task.status = 'failed'
-                task.job_explanation = 'Previous Task Failed: {"job_type": "%s", "job_name": "%s", "job_id": "%s"}' % (
-                    get_type_for_model(type(dep)),
-                    dep.name,
-                    dep.id,
-                )
-                task.save(update_fields=['status', 'job_explanation'])
-                task.websocket_emit_status('failed')
-                return dep
-
-        return None
+    def get_running_workflow_jobs(self):
+        graph_workflow_jobs = [wf for wf in WorkflowJob.objects.filter(status='running')]
+        return graph_workflow_jobs
 
     @timeit
-    def get_tasks(self, status_list=('pending', 'waiting', 'running')):
-        jobs = [j for j in Job.objects.filter(status__in=status_list).prefetch_related('instance_group')]
-        inventory_updates_qs = (
-            InventoryUpdate.objects.filter(status__in=status_list).exclude(source='file').prefetch_related('inventory_source', 'instance_group')
-        )
+    def get_tasks(self, filter_args):
+        jobs = [j for j in Job.objects.filter(**filter_args).prefetch_related('instance_group')]
+        inventory_updates_qs = InventoryUpdate.objects.filter(**filter_args).exclude(source='file').prefetch_related('inventory_source', 'instance_group')
         inventory_updates = [i for i in inventory_updates_qs]
         # Notice the job_type='check': we want to prevent implicit project updates from blocking our jobs.
-        project_updates = [p for p in ProjectUpdate.objects.filter(status__in=status_list, job_type='check').prefetch_related('instance_group')]
-        system_jobs = [s for s in SystemJob.objects.filter(status__in=status_list).prefetch_related('instance_group')]
-        ad_hoc_commands = [a for a in AdHocCommand.objects.filter(status__in=status_list).prefetch_related('instance_group')]
-        workflow_jobs = [w for w in WorkflowJob.objects.filter(status__in=status_list)]
+        project_updates = [p for p in ProjectUpdate.objects.filter(**filter_args).filter(job_type='check').prefetch_related('instance_group')]
+        system_jobs = [s for s in SystemJob.objects.filter(**filter_args).prefetch_related('instance_group')]
+        ad_hoc_commands = [a for a in AdHocCommand.objects.filter(**filter_args).prefetch_related('instance_group')]
+        workflow_jobs = [w for w in WorkflowJob.objects.filter(**filter_args)]
         all_tasks = sorted(jobs + project_updates + inventory_updates + system_jobs + ad_hoc_commands + workflow_jobs, key=lambda task: task.created)
+        logger.debug(f"{self.prefix} {all_tasks}")
         return all_tasks
```

Comment:
So get_tasks is still shared between the subclasses? That seems like a lot of unnecessary stuff, as the workflow manager will only use workflows, etc.

Reply:
currently the WorkflowManager doesn't call get_tasks, instead calls `get_running_workflow_jobs`

we could, for consistency, give WorkflowManager its own `get_tasks` which just returns get_running_workflow_jobs